core/src/main/java/org/apache/carbondata/core/carbon/datastore/chunk/reader/dimension/v2/CompressedDimensionChunkFileBasedReaderV2.java

```diff
@@ -135,6 +135,7 @@ public CompressedDimensionChunkFileBasedReaderV2(final BlockletInfo blockletInfo
       dimensionChunk = fileReader.readByteArray(filePath, dimensionChunksOffset.get(blockIndex),
           dimensionChunksLength.get(blockIndex));
       dimensionColumnChunk = CarbonUtil.readDataChunk(dimensionChunk);
+      assert dimensionColumnChunk != null;
```

Comment:
ok. And I found that FileHolder interface is ignoring IOException (catched but just log, not throwing it). So now I changed FileHolder and the caller to throw IOException